src/openloco/ui.cpp

```diff
@@ -776,22 +780,96 @@ namespace openloco::ui
         SDL_ShowSimpleMessageBox(SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, title.c_str(), message.c_str(), window);
     }
 
-    bool updateSDLResolution()
+    bool setDisplayMode(config::screen_mode mode, int16_t width, int16_t height)
     {
-        SDL_DisplayMode mode;
-        SDL_GetWindowDisplayMode(window, &mode);
+        SDL_DisplayMode DisplayMode;
+        SDL_DisplayMode DesktopDisplayMode;
+        Resolution DesktopResolution;
+        Resolution FallbackResolution;
```

Comment:
Please keep the code style in mind. Types should be TitleCase, but variables and functions should use camelCase. This makes it easier to distinguish between types and variables. (Some older functions still use snake_case, but never TitleCase.)

```suggestion
        SDL_DisplayMode displayMode;
        SDL_DisplayMode desktopDisplayMode;
        Resolution desktopResolution;
        Resolution fallbackResolution;
```